### ISSUE TITLE
conflict with other API using postmessage

### ISSUE THREAD

**Issue by lab9fr** [kylewelsby/angular-post-message#3]
Hello,
I've been using your API to communicate between a main page and an Iframe both running with angular.
On a sideway, I'm displaying videos using the youtube IFrame API, which make use of postMessage.
As soon as Youtube starts, ngPostMessage stops working.
I think it might have to do with this line :

``` javascript
$scope.sender = event.source;
```

My application communicating only through ngPostMessage, I added a verification system to be sure to only listen to messages emited by ngPostMessage and avoid other messages.

Here is the modified code :

``` javascript
(function() {
    var app;

    app = angular.module("ngPostMessage", ['ng']);

    app.directive('html', [
        '$window', '$postMessage', function($window, $postMessage) {
            return {
                restrict: 'E',
                controller: [
                    '$scope', function($scope) {
                        return $scope.$on('$messageOutgoing', function(event, message, domain) {
                            var sender;
                            if (domain == null) {
                                domain = "*";
                            }
                            sender = $scope.sender || $window.parent;
                            var msg = angular.toJson({module:'ngPostMessage', message:message});
                            return sender.postMessage(msg, domain);
                        });
                    }
                ],
                link: (function($scope, $element, $attrs) {
                    $scope.sendMessageToService = (function(event) {
                        var error, response;
                        event = event.originalEvent || event;
                        if (event && event.data) {
                            response = null;
                            try {
                                response = angular.fromJson(event.data);
                            } catch (_error) {
                                error = _error;
                                console.error('ahem', error);
                                response = event.data;
                            }

                            if (response.hasOwnProperty('module') && response.module == 'ngPostMessage')
                            {
                                $scope.sender = event.source;
                                $scope.$root.$broadcast('$messageIncoming', response.message);
                                return $postMessage.messages(response.message);
                            }

                        }
                    });
                    return angular.element($window).bind('message', $scope.sendMessageToService);
                })
            };
        }
    ]);

    app.factory("$postMessage", [
        '$rootScope', function($rootScope) {
            var $messages, api;
            $messages = [];
            api = {
                messages: (function(_message_) {
                    if (_message_) {
                        $messages.push(_message_);
                        $rootScope.$digest();
                    }
                    return $messages;
                }),
                lastMessage: (function() {
                    return $messages[$messages.length - 1];
                }),
                post: (function(message, domain) {
                    if (domain == null) {
                        domain = "*";
                    }
                    return $rootScope.$broadcast('$messageOutgoing', message, domain);
                })
            };
            return api;
        }
    ]);

}).call(this);
```

**Comment by kylewelsby** [comment]
Hi @lab9fr thanks for opening a issue ticket are you able to create a test case example to replicate this issue?

**Comment by kylewelsby** [comment]
hey @lab9fr I think #10 will solve your issue. I am closing this issue now.

**Comment by lab9fr** [comment]
Hi kylewelsby,

In the application I made, I had to communicate between an iframe and the page integrating the iframe. Both were using angular. But inside the iframe, I had to use youtube, which constantly post messages. 
And with the line 
<code>
$scope.sender = event.source;
</code>
The sender rapidely became Youtube, which prevented the communication to work after that.
but this was an extreme case, I found the workaround as provided in the code I've put earlier.

Main changes are to wrap the message sent in an object with an id to check on the other side : 
<code>
var msg = angular.toJson({module:'ngPostMessage', message:message});
</code>

and actually on the other side : 
<code>
if (response.hasOwnProperty('module') && response.module == 'ngPostMessage')
                            {
                                $scope.sender = event.source;
                                $scope.$root.$broadcast('$messageIncoming', response.message);
                                return $postMessage.messages(response.message);
                            }
</code>

Thought this might be useful if anyone encounter the same difficulties.
